spec/functions/assert_supported_architecture_spec.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+# TODO: test the error case, however due to an issue with boltspec
+# and functions we cannot do this right now.
+# https://github.com/puppetlabs/bolt/issues/1688
+
+describe 'peadm::assert_supported_architecture' do
+  let(:pre_condition) do
```

Comment:
@reidmv is this `pre_condition` being used?

Should we remove it?

Reply:
That's a question @davidsandilands might be able to answer with what he's learned about Bolt testing. The reason it's here is, back when this file was first written, `TargetSpec` was not a valid Puppet type, and so the pre_condition defined it for the purpose of the test.

It's possibly redundant now. I would suggest that changing it though is beyond the scope of this PR. The reason it's marked as a "new" change here is because the file has been renamed. The `assert_supported_architecture_spec.rb` file is really just the `validate_architecture_spec.rb` file renamed and slightly updated with the new function name and output.

Reply:
So I wouldn't overestimate my knowledge here :-) but I believe the boltspec cover this fully now and the mentioned bug ticket is closed so it could be removed in a seperate ticket.